core/dbt/events/serialization.py

```diff
@@ -0,0 +1,40 @@
+from mashumaro import DataClassDictMixin
+from mashumaro.config import (
+    BaseConfig as MashBaseConfig
+)
+from mashumaro.types import SerializationStrategy
+from datetime import datetime
+from dateutil.parser import parse
+from typing import cast
+
+
+class ExceptionSerialization(SerializationStrategy):
+    def serialize(self, value):
+        out = str(value)
+        return out
+
+    def deserialize(self, value):
+        return (Exception(value))
+
+
+class DateTimeSerialization(SerializationStrategy):
+    def serialize(self, value):
+        out = value.isoformat()
+        # Assume UTC if timezone is missing
+        if value.tzinfo is None:
+            out += "Z"
+        return out
+
+    def deserialize(self, value):
+        return (
+            value if isinstance(value, datetime) else parse(cast(str, value))
+        )
+
+
+class dbtClassEventMixin(DataClassDictMixin):
```

Comment:
can we change the name of this class? I know there's precedence for it, but I'd rather it be called `Serializable` or something more descriptive to why it's there and what it adds

Reply:
This name is in line with the other dbt mashumaro serialization class, which is dbtClassMixin.